pkg/cloud/services/networking/network.go

```diff
@@ -248,6 +253,39 @@ func (s *Service) createSubnet(openStackCluster *infrav1.OpenStackCluster, clust
 	return subnet, nil
 }
 
+// updateSubnetDNSNameservers updates the DNS nameservers for an existing subnet if they differ from the desired configuration.
+func (s *Service) updateSubnetDNSNameservers(openStackCluster *infrav1.OpenStackCluster, subnet *subnets.Subnet) error {
+	// Picking the first managed subnet since we only support one for now
+	desiredNameservers := openStackCluster.Spec.ManagedSubnets[0].DNSNameservers
+	currentNameservers := subnet.DNSNameservers
+
+	var needsUpdate bool
+	if len(desiredNameservers) != len(currentNameservers) {
+		needsUpdate = true
+	} else {
+		needsUpdate = !equality.Semantic.DeepEqual(currentNameservers, desiredNameservers)
+	}
```

Comment:
Probably not that big of a deal but I'm curious why don't we just drop `needsUpdate` and use the equality all the time.  Is this an optimization or is it simply something I don't know? :)

Reply:
`equality.Semantic.DeepEqual` is used in multiple places, so I guess it makes sense to have a wrapper function to compare slices, something like this

```
func EqualIgnoreOrder[T cmp.Ordered](a, b []T) bool {
	if len(a) != len(b) {
		return false
	}

	if len(a) == 0 {
		return true
	}

	aCopy := slices.Clone(a)
	bCopy := slices.Clone(b)

	slices.Sort(aCopy)
	slices.Sort(bCopy)

	return slices.Equal(aCopy, bCopy)
}
```

Reply:
This is just a nit, and mostly because I've spent time in the past reading Semantic.DeepEqual trying to determine the precise behaviour of its edge cases.

I only mention it at all because my personal preference would be to use something more explicit, especially when the data structure is simple as here. But this works and the PR has great test coverage, so there's no pressing need to update this.

Reply:
I am fine replacing if we bring similar helper in, I would hate to replace 1 func call with len,copy,sort,equal checks in multiple places, this can be followup if that makes sense